lib/plenario_web/controllers/api/utils.ex

```diff
@@ -254,4 +294,50 @@ defmodule PlenarioWeb.Api.Utils do
     |> resp(code, body)
     |> halt()
   end
+
+  # CONTROLLER HELPERS -- FETCHING, VALIDATING, QUERYING
+
+  @spec validate_data_set(String.t()) :: Plenario.Schemas.Meta.t() | :error | nil
+  def validate_data_set(slug), do: validate_data_set(slug, [])
+
+  @spec validate_data_set(String.t(), Keyword.t()) :: Plenario.Schemas.Meta.t() | :error | nil
+  def validate_data_set(slug, opts) when is_bitstring(slug) do
+    case Regex.match?(~r/^\d+$/, slug) do
+      true ->
+        nil
+
+      false ->
+        MetaActions.get(slug, opts)
+    end
+  end
+
+  def validate_data_set(_, _), do: :error
+
+  @spec apply_filter(Ecto.Queryable.t(), String.t(), String.t(), any()) :: Ecto.Queryable.t()
+  def apply_filter(query, fname, "lt", value), do: where(query, [q], field(q, ^fname) < ^value)
+
+  def apply_filter(query, fname, "le", value), do: where(query, [q], field(q, ^fname) <= ^value)
+
+  def apply_filter(query, fname, "eq", value), do: where(query, [q], field(q, ^fname) == ^value)
+
+  def apply_filter(query, fname, "ge", value), do: where(query, [q], field(q, ^fname) >= ^value)
+
+  def apply_filter(query, fname, "gt", value), do: where(query, [q], field(q, ^fname) > ^value)
+
+  def apply_filter(query, fname, "in", value), do: where(query, [q], field(q, ^fname) in ^value)
```

Comment:
These filters are just renaming the `where_condition` functions but you've completely deleted all the comments. Why? This shoots us in the foot in the future.